op-devstack/dsl/l2_el.go

```diff
@@ -85,3 +86,36 @@ func (el *L2ELNode) BlockRefByNumber(num uint64) eth.BlockRef {
 	el.require.NoError(err, "block not found using block number %d", num)
 	return block
 }
+
+// ReorgTriggered returns a lambda that checks that a L2 reorg occurred on the expected block
+// Composable with other lambdas to wait in parallel
+func (el *L2ELNode) ReorgTriggered(target eth.L2BlockRef, attempts int) CheckFunc {
+	return func() error {
+		el.log.Info("expecting chain to reorg on block ref", "id", el.inner.ID(), "chain", el.inner.ID().ChainID, "target", target)
+		return retry.Do0(el.ctx, attempts, &retry.FixedStrategy{Dur: 2 * time.Second},
+			func() error {
+				reorged, err := el.inner.EthClient().BlockRefByNumber(el.ctx, target.Number)
+				if err != nil {
+					if strings.Contains(err.Error(), "not found") { // reorg is happening wait a bit longer
+						el.log.Info("Supervisor still hasn't reorged chain A", "error", err)
```

Comment:
nit: There can be other issues which trigger L2 reorg, not just supervisor.

Reply:
You're right, I will modify the log line.